plugin/src/main/java/org/jenkinsci/plugins/structs/SymbolLookup.java

```diff
@@ -68,11 +70,55 @@
         }
     }
 
+    /**
+     * Looks for a {@link Descriptor} that has the given symbol
+     *
+     * @param type
+     *      Restrict the search to a subset of {@link Describable}
+     */
+    public <T> Descriptor<? extends T> findDescriptor(Class<T> type, String symbol) {
+        try {
+            Key k = new Key("findDescriptor",type,symbol);
+            Object i = cache.get(k);
+            if (i!=null)    return (Descriptor)i;
+
+            // not allowing @Symbol to use an invalid identifier.
+            // TODO: compile time check
+            if (!Utilities.isJavaIdentifier(symbol))
+                return null;
+
+            for (Class<?> e : Index.list(Symbol.class, pluginManager.uberClassLoader, Class.class)) {
+                if (Descriptor.class.isAssignableFrom(e)) {
```

Comment:
I guess we have no hope of lazy-loading this way. (SezPoz would make it possible.)